examples/cfq/syntax-based/grammar_test.py

```diff
@@ -34,33 +34,42 @@ def test_generate_grammar(self):
       filter_clause: "FILTER" "(" var_token "!=" var_token ")"
       triples_block: var_token TOKEN var_token
       var_token: VAR
-                | TOKEN 
+               | TOKEN 
       VAR: "?x0" | "?x1" | "?x2" | "?x3" | "?x4" | "?x5" 
       TOKEN: /[^\s]+/
     """
-    grammar_dict = generate_grammar(grammar_str)
-    expected_grammar_dict = {
-      'r0': ('query', 'select_query'),
-      'r1': ('select_query', 'select_clause "WHERE" "{" where_clause "}"'),
-      'r2': ('select_clause', '"SELECT" "DISTINCT" "?x0"'),
-      'r3': ('select_clause', '"SELECT" "count(*)"'),
-      'r4': ('where_clause', 'where_entry'),
-      'r5': ('where_clause', 'where_clause "." where_entry'),
-      'r6': ('where_entry', 'triples_block'),
-      'r7': ('where_entry', 'filter_clause'),
-      'r8': ('filter_clause', '"FILTER" "(" var_token "!=" var_token ")"'),
-      'r9': ('triples_block', 'var_token TOKEN var_token'),
-      'r10': ('var_token', 'VAR'),
-      'r11': ('var_token', 'TOKEN'),
-      'r12': ('VAR', '"?x0"'),
-      'r13': ('VAR', '"?x1"'),
-      'r14': ('VAR', '"?x2"'),
-      'r15': ('VAR', '"?x3"'),
-      'r16': ('VAR', '"?x4"'),
-      'r17': ('VAR', '"?x5"'),
-      'r18': ('TOKEN', '/[^\\s]+/')
+    grammar = Grammar(grammar_str)
+    expected_branches = [
+      RuleBranch(0,"select_query"),
+      RuleBranch(1,"select_clause \"WHERE\" \"{\" where_clause \"}\""),
```

Comment:
if you use single quotes outside you don't have to escape the inner double quotes:

```suggestion
      RuleBranch(1,'select_clause "WHERE" "{" where_clause "}"'),
```

Reply:
cool, I didn't know that